### PR TITLE
Customizing our Pull Request Template

### PR DESCRIPTION
**Pull request by CoderDake**
Splitting this into a separate PR based on Kenzie's feedback in this PR https://github.com/flutter/devtools/pull/4743#discussion_r1024169273

This template is copied from the [Flutter org template](https://github.com/flutter/.github/blob/main/PULL_REQUEST_TEMPLATE.md), and customized for our needs

# Template Preview Below
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or there is a reason for not adding tests.

![build.yaml badge]

If you need help, consider asking for advice on the #hackers-devexp-📐 channel on [Discord].

[Contributor Guide]: https://github.com/flutter/devtools/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://discord.com/channels/608014603317936148/608020249119555594
[build.yaml badge]: https://github.com/flutter/devtools/actions/workflows/build.yaml/badge.svg

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+- [ ] I listed at least one issue that this PR fixes in the description above.
+- [ ] I updated/added relevant documentation (doc comments with `///`).
+- [ ] I added new tests to check the change I am making, or there is a reason for not adding tests.
+- Existing and new test status: ![build.yaml badge]
+
+If you need help, consider asking for advice on the #hackers-devexp-📐 channel on [Discord].
+
+<!-- Links -->
+[Contributor Guide]: https://github.com/flutter/devtools/blob/master/CONTRIBUTING.md
+[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
```
Comment (kenzieschmoll) [inline .github/PULL_REQUEST_TEMPLATE.md:20]: I don't think this applies to us?

**Review by kenzieschmoll — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
+- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
+- [ ] I read the [Flutter Style Guide] _recently_, and have followed its advice.
+- [ ] I signed the [CLA].
+- [ ] I listed at least one issue that this PR fixes in the description above.
+- [ ] I updated/added relevant documentation (doc comments with `///`).
+- [ ] I added new tests to check the change I am making, or there is a reason for not adding tests.
+- Existing and new test status: ![build.yaml badge]
+
+If you need help, consider asking for advice on the #hackers-devexp-📐 channel on [Discord].
```
Comment (kenzieschmoll) [inline .github/PULL_REQUEST_TEMPLATE.md:16]: point users to #devtools

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+## Pre-launch Checklist
+
+- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
+- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
+- [ ] I read the [Flutter Style Guide] _recently_, and have followed its advice.
+- [ ] I signed the [CLA].
+- [ ] I listed at least one issue that this PR fixes in the description above.
+- [ ] I updated/added relevant documentation (doc comments with `///`).
+- [ ] I added new tests to check the change I am making, or there is a reason for not adding tests.
+- Existing and new test status: ![build.yaml badge]
```
Comment (kenzieschmoll) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: what is this for?

Comment (kenzieschmoll) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: can this be on a new line below the checklist? It looks kind of funny as is (indented with no bullet point)

Comment (CoderDake) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: agreed

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (kenzieschmoll) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: Point to the flutter chat wiki instead. You can say, "consider asking for help on Discord"
see link here: https://github.com/flutter/devtools/pull/4926/files#diff-2c22fff01f45b14288a3152294267b837c6f428009d4d1e0c09d23c64ba6382eR118

**Pull request merged**